spec/lib/tasks/migrate_skills_to_institutions_task_spec.rb

```diff
@@ -0,0 +1,57 @@
+# frozen_string_literal: true
+
+require 'rails_helper'
+
+RSpec.describe "migrate_skills_to_institutions", type: :rake do
+  describe 'move_skills' do
+    let(:institution) { create :institution }
+    let(:skill_un) { create :skill }
+    let(:skill_deux) { create :skill }
+    let!(:expert) { create :expert, antenne: create(:antenne, institution: institution), skills: [skill_un, skill_deux] }
+    let(:rake_task) { Rake::Task["migrate_skills_to_institutions:move_skills"] }
+
+    before { rake_task.invoke }
+
+    after { rake_task.reenable }
+
+    it 'create 2 InstitutionSubject' do
+      institution_subject_un = InstitutionSubject.find_by(institution: institution, subject: skill_un.subject, description: skill_un.title)
+      institution_subject_deux = InstitutionSubject.find_by(institution: institution, subject: skill_deux.subject, description: skill_deux.title)
+
+      expect(institution_subject_un).not_to be_nil
+      expect(institution_subject_deux).not_to be_nil
+    end
+
+    it 'create 2 expertSubject' do
+      insitution_subject_un = InstitutionSubject.first
+      insitution_subject_deux = InstitutionSubject.last
+
+      expert_subject_un = ExpertSubject.find_by(expert: expert, institution_subject: insitution_subject_un)
+      expert_subject_deux = ExpertSubject.find_by(expert: expert, institution_subject: insitution_subject_deux)
+
+      expect(expert_subject_un).not_to be_nil
+      expect(expert_subject_deux).not_to be_nil
+    end
+  end
+
+  describe 'add_subject_to_matches' do
+    let(:institution) { create :institution }
+    let(:subject) { create :subject }
+    let(:skill) { create :skill, subject: subject }
+    let(:match_with_skill) { create :match, skill: skill }
+    let(:match_without_skill) { create :match, skill: nil }
+    let(:rake_task) { Rake::Task["migrate_skills_to_institutions:add_subject_to_matches"] }
+
+    before { rake_task.invoke }
+
+    after { rake_task.reenable }
+
+    it 'add a subject to a match with skill' do
+      expect(match_with_skill.subject).not_to be_nil
+    end
+
+    it 'add a subject to a match without skill' do
+      expect(match_without_skill.subject).not_to be_nil
```

Comment:
Haha, ces deux tests passent par accident :D

`match_with_skill` et `match_without_skill` ne sont pas appelés avant les `expect`, ils n’existent donc pas quand la tache rake est invoquée. Par contre, comme `factories/matche.rb` donne un `subject` au `match`, ça marche forcément :D

---
Le subject affecté à `match_with_skill` devrait forcément être le subject du skill, on peut vérifier l’égalité plutôt que “not nil”. C’est aussi plus facile à comprendre.